04.ls/ls.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+COLUMN_NUMBER = 3
+
+def main
+  directory_files = Dir.children(ARGV[0] || '.').reject { |file| file[0] == '.' }.sort
+  max_length = directory_files.map(&:length).max
+  show_ls(directory_files, max_length)
+end
+
+def show_ls(directory_files, max_length)
+  split_files(directory_files).transpose.each do |row_files|
+    row_files.each { |col_file| col_file == row_files.last ? print(col_file) : printf("%-#{max_length}s\t", col_file) }
+    puts
+  end
+end
+
+def split_files(directory_files)
+  used_slice_number = (directory_files.length / COLUMN_NUMBER.to_f).ceil
+  division_files = directory_files.each_slice(used_slice_number).to_a
```

Comment:
```suggestion
  divided_files = directory_files.each_slice(used_slice_number).to_a
```

divided = 分割された 、の方が英語的には自然かもしれません。

Reply:
`divided_files`に変更しました。